[stephaneleong] iP

stephaneleong opened the pull request:
# Easton :writing_hand:

> “Create a short list of things you need to get done today.” - Matt D'Avella ([My Simple Productivity System](https://www.youtube.com/watch?v=BtiQvhQF8IA))

## Task Management Made Easy (Capabilities
+ _Create_ New Tasks
+ _Delete_ A Task
+ Keep _Track_ Of Your Tasks
+ _Search_ Your Tasks

## Different Kind Of Tasks
1. **Todos**
1. **Deadlines**
1. **Events**

## A TODO List For You
- [x] Download Easton
- [x] Create New Tasks
- [x] Become More Productive

If you want to use Easton in your own project:

```java
public static void main(String[] args) {
    new Easton("file.csv").run();
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

stephaneleong committed:
Rename Duke.java to Easton.java

stephaneleong committed:
Add greeting and exit

stephaneleong committed:
Add echo

stephaneleong committed:
Add support for list

stephaneleong committed:
Add support for add to list

stephaneleong committed:
Add Task.java

stephaneleong committed:
Rename itemList to taskArray

stephaneleong committed:
Add support for mark

stephaneleong committed:
Add ToDo.java

stephaneleong committed:
Add Deadline.java

stephaneleong committed:
Add Event.java

stephaneleong committed:
Extract support for list & mark into methods

stephaneleong committed:
Fix toString print bug

stephaneleong committed:
Add support for tracking todo, event & deadline

stephaneleong committed:
Change Task to abstract class

stephaneleong committed:
Modify test case

stephaneleong committed:
Fix print bug

stephaneleong committed:
Add Action.java

stephaneleong committed:
Add IllegalActionException

stephaneleong committed:
Add support to handle IllegalActionException

stephaneleong committed:
Add support for EmptyDescriptionException

stephaneleong committed:
Add support for InvalidIndexException

stephaneleong committed:
Add support for InvalidFormatException

stephaneleong committed:
Change Array to ArrayList

stephaneleong committed:
Add support for delete

stephaneleong committed:
Add retrieval from CSV file

stephaneleong committed:
Add string format to save in csv

stephaneleong committed:
Add support to save data to CSV file

stephaneleong committed:
Change error message to saving data

stephaneleong committed:
Add data directory

stephaneleong committed:
Merge branch 'branch-Level-7'

stephaneleong committed:
Add support for LocalDateTime

stephaneleong committed:
Add DateTimeFormatException

stephaneleong committed:
Add support for DateTimeFormatException

stephaneleong committed:
Merge branch 'branch-Level-8'

stephaneleong committed:
Change Easton to a non-static program

stephaneleong committed:
Extract support for storage

stephaneleong committed:
Integrate storage support

stephaneleong committed:
Add support for ui

stephaneleong committed:
Integrate ui support

stephaneleong committed:
Remove unused getFilePath method

stephaneleong committed:
Merge branch 'branch-A-MoreOOP'

stephaneleong committed:
Move files to easton package

stephaneleong committed:
Move tasks to model package

stephaneleong committed:
Change to public class

stephaneleong committed:
Move ui to view package

stephaneleong committed:
fix bug

stephaneleong committed:
Merge branch 'branch-A-Packages'

stephaneleong committed:
Merge branch 'add-gradle-support'

stephaneleong committed:
Edit gradle configuration to run Easton

stephaneleong committed:
Add test case for createToDo

stephaneleong committed:
Fix bug in createToDo

stephaneleong committed:
Add test cases for createDeadline

stephaneleong committed:
Fix bug for createDeadline

stephaneleong committed:
Fix bug for ui welcome and goodbye

stephaneleong committed:
Add test case for Storage

stephaneleong committed:
Add archiveFileName to gradle build.gradle

stephaneleong committed:
Change methods to public for test

stephaneleong committed:
Add JavaDocs for public methods and classes

stephaneleong committed:
Rephrase JavaDoc for constructor

stephaneleong committed:
Add JavaDoc for exception

stephaneleong committed:
Fix JavaDoc capitalization

stephaneleong committed:
Add JavaDoc to public methods and class

stephaneleong committed:
Add JavaDoc to public methods and class

stephaneleong committed:
Add JavaDoc to public methods and class

stephaneleong committed:
Add Checkstyle to gradle build

stephaneleong committed:
Fix coding standard

weijianwong commented on src/main/java/easton/Easton.java:
```diff
@@ -0,0 +1,263 @@
+import java.io.*;
+import java.util.ArrayList;
+
+public class Easton {
```
Perhaps you could include javadoc header comments for classes and methods so that readers can better understand your code.

stephaneleong replied:
Hi @weijianwong! yes I was late in pushing the javadoc comments, I have included it for all my public classes and methods.

weijianwong commented on src/main/java/easton/Easton.java:
```diff
@@ -0,0 +1,263 @@
+                action = Action.INVALID;
+            }
+
+            switch (action) {
```
Great job with the indentation of the switch statement. Many students often overlook this part so good job for avoiding this mistake!

weijianwong commented on src/main/java/easton/Easton.java:
```diff
@@ -0,0 +1,263 @@
+        new Easton("task.csv").run();
+    }
+
+    private void changeTaskStatus(String input, boolean isDone, String message) {
```
Maybe you could move this logic to the Task class?

stephaneleong replied:
Okay I will look into this!

stephaneleong replied:
@weijianwong looking into the logic of this function, changeTaskStatus(). I'm worried if I shifted the logic into the Task class, specfically the printing of statement, that I would create a dependency assiocation between Task and the Ui classes which I forsee the code becoming more buggy if one of the parts were to break. But, yes input is appreciated.

weijianwong commented on src/main/java/easton/DateTimeFormatException.java:
```diff
@@ -0,0 +1,7 @@
+package easton;
+
+public class DateTimeFormatException extends Exception {
```
Perhaps this could be in another package such as the exceptions package?

stephaneleong replied:
okay. I will move all the exception into an exception package

weijianwong commented on src/main/java/easton/Storage.java:
```diff
@@ -0,0 +1,61 @@
+        this.filePath = filePath;
+    }
+
+    public ArrayList<String> retrieve() throws IOException {
```
Perhaps you could consider changing the naming of the functions in this class to make it clearer? You could name this function "retrieveRecord" instead of just "retrieve"?

stephaneleong replied:
Made the changes! thank you

weijianwong commented on src/main/java/easton/view/Ui.java:
```diff
@@ -0,0 +1,60 @@
+import java.util.ArrayList;
+import java.util.Scanner;
+
+public class Ui<T> {
```
Why does this class have to be a generic type?

stephaneleong replied:
Well I agree it is not necessary because of the scale of this ip, however, when I was planning out the design I thought it would also make sense to make this class generic so that if there were other Classes we would want to display (e.g. listing) I would not need to write code. 

However, I would appreciate your input on this thought of my @weijianwong. Thank you!

weijianwong commented on src/main/java/easton/Easton.java:
```diff
@@ -0,0 +1,263 @@
+package easton;
```
Package naming is in lower case. Class names are in PascalCase and variable names are in camelCase. Great job for following the naming conventions.

weijianwong reviewed (commented):
Overall great job for following the coding style conventions with the correct naming conventions and indentations. One improvement that i could suggest is to write header comments for all your classes and methods to make it clearer for the readers. But overall great job!

stephaneleong committed:
Add find to action

stephaneleong committed:
Add support to find keyword

stephaneleong committed:
Merge branch 'branch-A-JavaDoc'

stephaneleong committed:
Resolve merge conflict

stephaneleong committed:
Merge branch 'branch-A-CodingStandard'

stephaneleong committed:
Resolve merge conflict

stephaneleong committed:
Add Javadoc to method findTasks

stephaneleong committed:
Merge branch 'branch-Level-9'

stephaneleong committed:
Fix findTasks list bug

stephaneleong committed:
Move exceptions into a package

stephaneleong committed:
Rename Storage methods

stephaneleong committed:
Add support for multiple keywords to find

hazzle23 commented on src/main/java/easton/Easton.java:
```diff
@@ -0,0 +1,396 @@
+     * @param isDone Is the task done.
+     * @param message Message displayed to the user interface.
+     */
+    public void changeTaskStatus(String input, boolean isDone, String message) {
```
I like how you use clear descriptive verbs for all of the methods in this class. Good job on following the naming convention!

hazzle23 commented on src/main/java/easton/Action.java:
```diff
@@ -0,0 +1,8 @@
+package easton;
+
+/**
+ * Represents that the chatbot can handle.
```
I think the JavaDoc could be more descriptive to better explain what `Action` represents. A possible revision could be 
`*Represents the different types of actions/commands that the chatbot can handle.`

hazzle23 commented on src/main/java/easton/model/Event.java:
```diff
@@ -0,0 +1,48 @@
+public class Event extends Task {
+
+    private LocalDateTime from;
+    private LocalDateTime to;
```
While variable name `to` is functional, I think you can consider renaming it to something more descriptive for better clarity, like `endDate` or `toDate`.

hazzle23 commented on src/test/java/easton/EastonTest.java:
```diff
@@ -0,0 +1,60 @@
+import easton.exception.InvalidFormatException;
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.assertThrows;
```
Based on [this guide](https://se-education.org/guides/tutorials/intellijCodeStyle.html#tweak-import-order), I think static imports should be ordered first.

hazzle23 commented on src/test/java/easton/EastonTest.java:
```diff
@@ -0,0 +1,60 @@
+    }
+
+    @Test
+    public void createDeadline_incorrectDateTimeFormat_exceptionThrown() {
```
Good job on following the test methods naming convention!

hazzle23 reviewed (commented):
Overall looks good to me!

stephaneleong committed:
Fix coding standards for tests

stephaneleong committed:
Add JavaFX to gradle

stephaneleong committed:
Change display to console in run method

stephaneleong committed:
Add support for Easton response

stephaneleong committed:
Add support for GUI

stephaneleong committed:
Merge branch 'branch-Level-10'

stephaneleong committed:
Rename DateTime variables

stephaneleong committed:
Change Class Javadoc

stephaneleong committed:
Add assertion

No assertion has been implemented in the project.

Developers can create instances without assigning proper values
being assigned. Possibly, running into errors when using other methods.

Add not null assertion to ensure parameters is assigned a proper value.
Add not blank assertion to ensure String is not blank.

By adding these assertion, developers would know that the program
is not designed for these values when they execute the project.

stephaneleong committed:
Merge pull request #1 from stephaneleong/branch-A-Assertions

Add assertion

stephaneleong committed:
Refactor code base to improve readability

Many methods within the code base are long and does not follow the SLAP
principle. Arrowhead style code is common in methods.

Code is unreadable making it hard for other developers to understand and
 follow the logic of each function.

Simplify complicated expressions.
Avoid Magic Literals.
Extract into smaller functions.
Minimise scope of variables.

By applying all of these principles. Code logic is easier to follow for
developers.

stephaneleong committed:
Resolve merge conflict with master

stephaneleong committed:
Merge pull request #2 from stephaneleong/branch-A-CodeQuality

Branch a code quality

stephaneleong committed:
Delete IllegalActionException.java

stephaneleong committed:
Organise methods

stephaneleong committed:
Adjust test to new external behaviour

stephaneleong committed:
Add new parse exception for task

stephaneleong committed:
Extract parse method from retrieveTasks

stephaneleong committed:
Fix Javadoc comments convention

stephaneleong committed:
Add support for archive

stephaneleong committed:
Merge branch 'branch-C-Archive'

stephaneleong committed:
Add test cases to saveRecords

stephaneleong committed:
Change text displayed

stephaneleong committed:
Update docs for user guide
